src/main/java/org/ehealth/dhis2SmsComp/SmsDecoder.java

```diff
@@ -43,7 +44,13 @@ public SmsDecoder(ArrayList<SmsCommand> smsCmds) {
 	 * @throws Exception 
 	 */
 	public String decode(byte[] smsBytes) throws Exception {
-		SmsSubmission subm = decodeSMS(smsBytes);
+		SmsSubmission subm = null;
+		try {
```

Comment:
Tried to keep this handling as simple as possible so as not to lose track of odd cases but I'm open to considering more sophisticated solutions if there are obvious problems.